client/src/controllers/DismissibleController.ts

```diff
@@ -0,0 +1,56 @@
+import { Controller } from '@hotwired/stimulus';
+import { WAGTAIL_CONFIG } from '../config/wagtailConfig';
+
+/**
+ * <section
+ *  aria-labelledby="whats-new-heading"
+ *  data-controller="w-dismissible"
+ *  data-w-dismissible-dismissed-class="w-dismissible--dismissed"
+ *  data-w-dismissible-id-value="{{ dismissible_id }}"
+ * >
+ * <div>
+ *  <button
+ *      type="button"
+ *      data-action="w-dismiss#dismissible"
+ *  >
+ *      <span class="w-sr-only">{% trans "Close" %}</span>
+ *  </button>
+ *  </div>
+ * </section>
+ */
+export class DismissibleController extends Controller<HTMLElement> {
+  static classes = ['dismissed'];
+
+  static values = {
+    id: { default: '', type: String },
+    dismissed: { default: false, type: Boolean },
+  };
+
+  declare dismissedClass: string;
+  declare idValue: string;
+  declare dismissedValue: boolean;
+
+  /**
+   * Initialise dismissibles fetched from server and add appropriate attributes to them on click.
+   */
+  toggle(): void {
+    this.element.classList.add(this.dismissedClass);
+    this.dismissedValue = true;
+    (this.constructor as typeof DismissibleController).updateDismissibles({
+      [this.idValue]: true,
+    });
+  }
+
+  static updateDismissibles(data: Record<string, boolean>): Promise<Response> {
```

Comment:
the more I see this, I think we should actually just make this an exported named function (can be in the same file), not a static method.

Sorry for the run-around here, it's hard to know what do to until you see the code sometimes.

Reply:
Sure!
That gives us more leeway to access the method